api/leaderworkerset/v1/leaderworkerset_types.go

```diff
@@ -176,6 +183,31 @@ type SubGroupPolicy struct {
 	SubGroupSize *int32 `json:"subGroupSize,omitempty"`
 }
 
+type NetworkConfig struct {
+	// SubdomainPolicy determines the policy that will be used when creating
+	// the headless service
+	SubdomainPolicy SubdomainPolicy `json:"subdomainPolicy"`
+}
+
+type SubdomainPolicy string
+
+const (
+	// SubdomainShared will create a single headless service that all replicas
+	// will share. The host names look like:
+	// Replica 0: my-lws-0.my-lws, my-lws-0-1.my-lws
+	// Replica 1: my-lws-1.my-lws, my-lws-1-1.my-lws
+	SubdomainShared SubdomainPolicy = "Shared"
+	// SubdomainLeadersSharedWorkersDedicated will create a headless service for the leaders,
+	// and a headless service per replica for the workers.
+	// The leader host names will look like:
+	// Replica 0: my-lws-0.my-lws
+	// Replica 1: my-lws-1.my-lws
+	// The worker host names will look like:
+	// Replica 0: my-lws-0-1.my-lws-0, my-lws-0-2.my-lws-0
```

Comment:
if we switched from "Shared" to "LeadersSharedWorkersDedicated", the worker pods' hostnames will change  right? for tpu case, do we need to update the container env vars as well? and do we need to trigger a rolling update?

Reply:
That makes sense, they will need to be updated. In the case where there is an update from "Shared" to "LeadersSharedWorkersDedicated", and one of the TPUs disconnects but doesn't cause a restart, when the TPU attempts to connect again the `TPU_WORKER_HOSTNAMES` env variable will have the wrong value, causing it to fail. We can trigger a rolling update, which in the default case has `MaxSurge=0` and `MaxUnavailable=1`, so updating it won't cause much disruption.